### DISCUSSION TITLE
Physics: removing a Solid or changing a Solid's physical shape?

### DISCUSSION THREAD

**Discussion by jackdaus** [StereoKit/StereoKit#408] [Q&A] [answered]
Two questions about StereoKit's physics features:

1) Once I create a Solid, is there a way I can remove the solid from the physics system?

2) I see in the docs that I can change a Solid's physical shape by calling AddBox, AddSphere, etc. But is there a way I can remove these shapes once added? Maybe a way I can reset the Solid's physical shape?

Here's my use case in case: I've got one spider I'm rendering on the screen but sometimes I want to swap which model I am using for the spider. When I swap out the model, I also want to change the dimensions of the solid to match the new model. 

Thanks!

**Comment by maluoi** [comment, marked as answer]
Oh, actually this part of the physics code looks a bit broken! I'm not sure how I didn't notice this before, but from the C# side, it never actually removes items from the physics system, even when it's nulled out and deconstructed! 😬 I just fixed _that_ bit, and it'll be in the next update.

I'll look into adding a function for resetting the shapes, that should be pretty easily doable! You can also set `Solid.Enabled = false` to remove it from the physics simulation right away as well. Note that the deconstructor only gets called at Garbage Collection time, so setting `Enabled = false` is probably a good approach to take when getting rid of a Solid regardless!

**Reply by jackdaus** [reply]
Awesome thanks! I thought it was funny that my solids seemed to stick around even after there were no references to them 😅

I may end up going with a differnet design to avoid the need of resetting the solid's shape. I'm thinking I could create a solid for each model at init time. And then enable the one solid currently being used and disable the other solids. But I think a shape reset function could still be useful in general!
